src/DurableSDK/PowerShellServices.cs

```diff
@@ -6,50 +6,199 @@
 namespace Microsoft.Azure.Functions.PowerShellWorker.Durable
 {
     using System;
+    using System.Collections.ObjectModel;
+    using System.Linq;
     using System.Management.Automation;
-    using PowerShell;
+    using Microsoft.Azure.Functions.PowerShellWorker.PowerShell;
+    using Microsoft.Azure.Functions.PowerShellWorker.Utility;
+    using Microsoft.Azure.WebJobs.Script.Grpc.Messages;
+    using Newtonsoft.Json;
+    using LogLevel = WebJobs.Script.Grpc.Messages.RpcLog.Types.Level;
 
     internal class PowerShellServices : IPowerShellServices
     {
-        private const string SetFunctionInvocationContextCommand =
-            "Microsoft.Azure.Functions.PowerShellWorker\\Set-FunctionInvocationContext";
-
         private readonly PowerShell _pwsh;
-        private bool _hasSetOrchestrationContext = false;
+        private bool _hasInitializedDurableFunctions = false;
+        private bool _usesExternalDurableSDK = false;
+        private readonly ErrorRecordFormatter _errorRecordFormatter = new ErrorRecordFormatter();
+        private readonly ILogger _logger;
+
+        // uses built-in SDK by default
+        private string SetFunctionInvocationContextCommand = string.Format(
+            PowerShellWorkerStrings.SetFunctionInvocationContextCmdLetTemplate,
+            PowerShellWorkerStrings.InternalDurableSDKName);
 
-        public PowerShellServices(PowerShell pwsh)
+        public PowerShellServices(PowerShell pwsh, ILogger logger)
         {
             _pwsh = pwsh;
+            _logger = logger;
+
+            // Configure FunctionInvocationContext command, based on the select DF SDK
+            var prefix = PowerShellWorkerStrings.InternalDurableSDKName;
+            SetFunctionInvocationContextCommand = string.Format(PowerShellWorkerStrings.SetFunctionInvocationContextCmdLetTemplate, prefix);
+        }
+
+        private bool tryImportingDurableSDK()
+        {
+            // Try to load/import the external Durable Functions SDK. If an error occurs, it is logged.
+            var importSucceeded = false;
+            try
+            {
+                // attempt to import SDK
+                _logger.Log(isUserOnlyLog: false, LogLevel.Trace, String.Format(
+                    PowerShellWorkerStrings.LoadingDurableSDK, PowerShellWorkerStrings.ExternalDurableSDKName));
+
+                var results = _pwsh.AddCommand(Utils.ImportModuleCmdletInfo)
+                    .AddParameter("FullyQualifiedName", PowerShellWorkerStrings.ExternalDurableSDKName)
+                    .AddParameter("ErrorAction", ActionPreference.Stop)
+                    .AddParameter("PassThru")
+                    .InvokeAndClearCommands<PSModuleInfo>();
+
+                var numResults = results.Count();
```

Comment:
I believe that only one PowerShell module can be loaded at the time (we should test this to make sure). If this is the case, then the imported module result should always be 1. 

Also, if the result were more than one, it would be helpful if we log the module name, version, and path in Kusto for CRI investigations.

Reply:
Yes, I think you're right about the result always being 1. I was being overly defensive with that if-statement check. I also made a local test which validated our expectations.

I'll be removing this check then!